File: rpnumerics/physics/PermParams.java
/*
 * Instituto de Matematica Pura e Aplicada - IMPA
 * Departamento de Dinamica dos Fluidos
 *
 */

package rpnumerics.physics;

public class PermParams {
    static public final double DEFAULT_LW = 0d;
    static public final double DEFAULT_LOW = 0d;
    static public final double DEFAULT_LG = 0d;
    static public final double DEFAULT_LOG = 0d;
    static public final double DEFAULT_CNW = 0d;
    static public final double DEFAULT_CNO = 0d;
    static public final double DEFAULT_CNG = 0d;
    static public final double DEFAULT_EPSL = 0d;
    private double lw_, low_, log_, lg_;
    private double cnw_, cno_, cng_;
    private double epsl_;

    public PermParams(double lw, double low, double lg, double log, double cnw, double cno, double cng, double epsl) {
        lw_ = lw; low_ = low;
        lg_ = lg; log_ = log;
        cnw_ = cnw; cno_ = cno; cng_ = cng;
        epsl_ = epsl;
    }

    public PermParams() { reset(); }

    public double lw() { return lw_; }

    public double low() { return low_; }

    public double lg() { return lg_; }

    public double log() { return log_; }

    public double cnw() { return cnw_; }

    public double cno() { return cno_; }

    public double cng() { return cng_; }

    public double epsl() { return epsl_; }

    public void reset() {
        lw_ = DEFAULT_LW;
        low_ = DEFAULT_LOW;
        lg_ = DEFAULT_LG;
        log_ = DEFAULT_LOG;
        cnw_ = DEFAULT_CNW;
        cno_ = DEFAULT_CNO;
        cng_ = DEFAULT_CNG;
        epsl_ = DEFAULT_EPSL;
    }
}

File: src/rpnumerics/physics/triphase/PermParams.cc
/*
 * IMPA - Fluid Dynamics Laboratory
 *
 * RPn Project
 *
 * @(#) PermParams.cc
 */

/*
 * ---------------------------------------------------------------
 * Includes:
 */
#include "PermParams.h"

/*
 * ---------------------------------------------------------------
 * Definitions:
 */


const double PermParams::DEFAULT_LW = 0;
const  double PermParams:: DEFAULT_LOW = 0;
const  double PermParams:: DEFAULT_LG = 0;
const  double PermParams::DEFAULT_LOG = 0;
const  double PermParams::DEFAULT_CNW = 0;
const  double PermParams::DEFAULT_CNO = 0;
const  double PermParams::DEFAULT_CNG = 0;
const  double PermParams:: DEFAULT_EPSL = 0;





PermParams::PermParams(double lw, double low, double lg, double log, double cnw, double cno, double cng, double epsl) {
    lw_ = lw; low_ = low;
    lg_ = lg; log_ = log;
    cnw_ = cnw; cno_ = cno; cng_ = cng;
    epsl_ = epsl;
}
PermParams::PermParams() { reset(); }

void PermParams::reset() {
    lw_ = DEFAULT_LW;
    low_ = DEFAULT_LOW;
    lg_ = DEFAULT_LG;
    log_ = DEFAULT_LOG;
    cnw_ = DEFAULT_CNW;
    cno_ = DEFAULT_CNO;
    cng_ = DEFAULT_CNG;
    epsl_ = DEFAULT_EPSL;
}

//! Code comes here! daniel@impa.br

